sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/utils.scala

```diff
@@ -154,143 +146,150 @@ object Utils {
     val aggregateExpressions = functionsWithDistinct ++ functionsWithoutDistinct
     val usesTungstenAggregate =
       child.sqlContext.conf.unsafeEnabled &&
-        aggregateExpressions.forall(
-          _.aggregateFunction.isInstanceOf[DeclarativeAggregate]) &&
```

Comment:
Perhaps. Let me see if I'm convinced that the other tests are still valid even if we skip this distinct case.